src/mc-fle2-encryption-placeholder.c

```diff
@@ -282,13 +300,24 @@ static bool mc_FLE2RangeFindSpecEdgesInfo_parse(mc_FLE2RangeFindSpecEdgesInfo_t
     CHECK_HAS(indexMax)
     // Do not error if precision is not present. Precision optional and only
     // applies to double/decimal128.
+
+    if (out->trimFactor.set && !use_range_v2) {
+        // Once `use_range_v2` is default true, this block may be removed.
+        CLIENT_ERR("invalid FLE2RangeFindSpecEdgesInfo: 'trimFactor' is not supported for QE range v1");
+        return false;
+    }
+    // Do not error if trim factor is not present. It is optional as it requires a feature flag,
```

Comment:
It should be required once you land the server changes. There will never be a case where we don't send it to the server for range even when it is just 0.

Reply:
see above